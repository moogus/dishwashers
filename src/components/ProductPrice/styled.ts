import styled from 'styled-components';

export const PriceText = styled.p`
  font-size: 32px;
  text-align: left;
  color: #444;
  margin: 0;
`;

export const TextMarginContainer = styled.div`
  margin-top: 30px;
  margin-left: 10px;
  @media (max-width: 768px) {
    margin-left: 5%;
    margin-right: 5%;
    margin-bottom: 40px;
  }
`;

export const RedText = styled.p`
  color: red;
  margin-bottom: 0;
  max-width: 250px;
`;
