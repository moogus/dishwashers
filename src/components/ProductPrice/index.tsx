import React from "react";

import useGetProduct from "../../hooks/product/get-product";

import { PriceText, TextMarginContainer, RedText } from "./styled";

export const ProductPrice = () => {
  const { price, additionalServices } = useGetProduct();
  const { includedServices = [] } = additionalServices;

  return (
    <TextMarginContainer className="row">
      <div className="col">
        <PriceText>{price}</PriceText>
        <RedText>Claim an extra 3 years guarantee via redemption</RedText>
        {includedServices.map((s) => (
          <p key={s}>{s}</p>
        ))}
      </div>
    </TextMarginContainer>
  );
};
