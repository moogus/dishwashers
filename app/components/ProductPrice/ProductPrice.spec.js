import React from 'react';
import { render } from '@testing-library/react';

import ProductPrice from 'components/ProductPrice';
import useGetProduct from 'hooks/product/get-product';

jest.mock('hooks/product/get-product');

describe('<ProductPrice/>', () => {
  const renderContainer = () => render(<ProductPrice />);

  describe('when <ProductPrice/> is rendered', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useGetProduct.mockReturnValue({
        price: 'Â£10000.00', additionalServices: ['some amazing stuff'],
      });
    });

    test('component renders as expected', () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });
});
