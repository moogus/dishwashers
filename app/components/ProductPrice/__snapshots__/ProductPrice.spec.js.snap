// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPrice/> when <ProductPrice/> is rendered component renders as expected 1`] = `
<div>
  <div
    class="styled__TextMarginContainer-om15ht-1 iGroMy row"
  >
    <div
      class="col"
    >
      <p
        class="styled__PriceText-om15ht-0 Lzojg"
      >
        Â£10000.00
      </p>
      <p
        class="styled__RedText-om15ht-2 kydASL"
      >
        Claim an extra 3 years guarantee via redemption
      </p>
    </div>
  </div>
</div>
`;
